cmd/server/app/serve.go

```diff
@@ -176,6 +182,9 @@ var serveCmd = &cobra.Command{
 		// Wait for all entity events to be executed
 		exec.Wait()
 
+		// Wait for the client cache background routine to finish
+		restClientCache.Wait()
+
```

Comment:
@jhrozek Question: According to this logic, if the go routines above finish their execution (somehow, though it shouldn't happen for a server), this would mean the context has not been cancelled, and this would lead to an infinite wait as the ticker only stops using defer and context cancellation. 

On the other hand, the go routines started above should keep running as it's a server binary. In case of panic, the whole process would be stopped. Just double-checking, whether this is correct or not :)